TELCODOCS-655: D/S Docs & RN: AI SaaS - Installing with the AI - Updates to existing content (#46991)
openshift/openshift-docs: modules/assisted-installer-how-to-use-the-assisted-installer.adoc

// This is included in the following assemblies:
//
// installing-on-prem-assisted.adoc
:_content-type: CONCEPT

[id='how-to-use-the-assisted-installer_{context}']
= How to use the Assisted Installer

> {product-title} {ai-full} is a user-friendly installation solution offered on link:http://console.redhat.com[console.redhat.com]. {ai-full} supports the various deployment platforms with a focus on bare metal. It comprises two key components:

Reviewer: I'd link here the actual AI link you added at the bottom to access AI directly.

Also, when referring to cloud.redhat.com I'd call it as Red Hat Hybrid Cloud Console

Author: I've added both to the sentence.

* *Assisted Service:* The Assisted Service provides for the hosting of installation artifacts, including:
  - Ignition files
  - The installation configuration
  - A discovery ISO
  - The web user interface
  - The REST API

* *{ai-full}*: The {ai-full} provides the installation functionality, and it has following advantages:
  - *Web user interface:* The web user interface performs cluster installation without the user having to create the installation configuration file manually.
  - *No bootstrap node:* A bootstrap node is not required when installing with the {ai-full}. The bootstrapping process executes on a random node within the cluster.

>   - *Simplified deployment:* Deployment does not require in-depth knowledge of {product-title}.

Reviewer 2: I would add also "rich pre flight validation and reasonable defaults"

Reviewer 3: what about mentioning a few of the main features:
SNO
vsphere
dual stack
static ip

Author: I've added those comments. Let me know what you think.

  - A flexible API
  - You can deploy Single Node OpenShift (SNO)
  - You can install {VirtProductName} and {rh-storage} (formerly OpenShift Container Storage) from the web user interface.

The {ai-full} is a great solution for:

* Users installing {product-title} on premises in a connected environment.
* Users without the need for strictly reproducible results. Developers frequently update the installable versions to the latest tested z-stream releases. Some older versions remain available, if needed.

> * Users that don't need automation. {ai-full} is great for first clusters, demos, and proof of concept projects.

Reviewer 4: I agree. This is not the way we want to position AI. Good catch. Users can (and do) build automation around ai apis. 

Same being said about 'no need for strictly repoducible results' - where does it comes from?

> * Users that don't need automation. {ai-full} is great for first clusters, demos, and proof of concept projects.

Reviewer 5: Instead of "Users that don't need automation", maybe you meant "users that don't have automation".  So it's great for first clusters, demos, PoCs as you said, but you can definitely add automation as the comments above have said.

> * Users that don't need automation. {ai-full} is great for first clusters, demos, and proof of concept projects.

Author: @jelkosz I've updated the doc to make it clearer that you can use the API to automate later so it doesn't sound like AI is just for trivial clusters.

// This is included in the following assemblies:
//
// installing-on-prem-assisted.adoc
:_content-type: CONCEPT

[id="how-to-use-the-assisted-installer_{context}"]

> = How to use the Assisted Installer

Reviewer 5: This doc isn't really about "how" to use the Assisted Installer, more like when or why.  Maybe just "Using the Assisted Installer"?

> The {product-title} link:https://console.redhat.com/openshift/assisted-installer/clusters/~new[{ai-full}] is a user-friendly installation solution offered on the link:http://console.redhat.com[Red Hat Hybrid Cloud Console]. The {ai-full} supports the various deployment platforms with a focus on bare metal.

Reviewer 5: @romfreiman WDYT?

```suggestion
The {product-title} link:https://console.redhat.com/openshift/assisted-installer/clusters/~new[{ai-full}] is a user-friendly installation solution offered on the link:http://console.redhat.com[Red Hat Hybrid Cloud Console]. The {ai-full} supports the various deployment platforms with a focus on bare metal and vSphere.
```

Author: Done.

> The {ai-full} provides installation functionality as a service. This software-as-a-service (SaaS)  approach has the following advantages:

Reviewer 5: ```suggestion
The {ai-full} provides installation functionality as a service. This software-as-a-service (SaaS) approach has the following advantages:
```

Author: Fixed.

* *Web user interface:* The web user interface performs cluster installation without the user having to create the installation configuration files manually.
* *No bootstrap node:* A bootstrap node is not required when installing with the {ai-full}. The bootstrapping process executes on a node within the cluster.
* *Hosting:* The {ai-full} hosts:
  - Ignition files
  - The installation configuration
  - A discovery ISO
  - The installer

> * *Simplified deployment:* Deployment does not require in-depth knowledge of {product-title}. The {ai-full} provides reasonable defaults. Additionally, the {ai-full} performs rich pre-flight validation before installation to catch configuration-related installation errors before they occur.

Reviewer 5: ```suggestion
* *Simplified deployment:* Deployment does not require in-depth knowledge of {product-title}. The {ai-full} provides reasonable defaults. Additionally, the {ai-full} performs rich pre-flight validations before installation to catch configuration-related installation errors before they occur.
```

Author: Fixed, but I consolidated this with streamlined workflow bullet.

* *Advanced networking:* The {ai-full} supports IPv4/IPv6 dual stack networking, NMState-based static IP addressing, and an HTTP/S proxy.

> * *REST API:* The {ai-full} has a REST API, enabling automation.

Reviewer 5: Do you want to add some links?
- This is the swagger that defines the API: https://github.com/openshift/assisted-service/blob/master/swagger.yaml
- Doc on how to use it: https://github.com/openshift/assisted-service/blob/master/docs/cloud.md

Author: They ask us not to link to github docs. So we'd probably just bring these downstream at some point.

The {ai-full} is a great solution for:

* *Pre-installation validation:* The {ai-full} validates the configuration before installation to ensure a high probability of success. Validation includes:
  - Ensuring network connectivity
  - Ensuring sufficient network bandwidth
  - Ensuring connectivity to the registry
  - Ensuring time synchronization between cluster nodes
  - Verifying that the cluster nodes meet the minimum hardware requirements
  - Validating the installation configuration parameters

> * *Multiple installation scenarios:* The {ai-full} supports numerous scenarios, including:
>   - Installing {product-title} on premises in a connected environment, including with an optional HTTP/S proxy.
>   - Installing Single Node OpenShift (SNO)
>   - Installing on vSphere
>   - Installing {VirtProductName} and {rh-storage} (formerly OpenShift Container Storage) from the web user interface
>   - Installing first clusters, demos, and proof of concept projects that don't have automation. Users can add automation to the {ai-full} through the API later

Reviewer 5: ```suggestion
The {ai-full} supports installing {product-title} on premises in a connected environment, including with an optional HTTP/S proxy.  It can install:
  - Highly available {product-title} or Single Node OpenShift (SNO)
  - Bare-metal or vSphere
  - Optionally installing {VirtProductName} and {rh-storage} (formerly OpenShift Container Storage) from the web user interface
  
The user interface provides an intuitive interactive flow where automation does not exist or is not required (e.g., first clusters, demos, and proof of concept projects). Users may also automate installations using the REST API.
```

Author: I made the changes. However, I omitted the exempli gratia, because it sort of re-introduces the idea that AI isn't suitable for production clusters. I think it reads fine without those examples. What do you folks think?

* *Streamlined installation workflow:* The {ai-full} provides the installer as a service, which:
  - Eliminates the need to install and run the installer locally
  - Ensures the latest version of the installer up to the latest tested z-stream releases. Older versions remain available, if needed

>   - Enables building automation through the API without the need to run the installer locally

Reviewer 5: These 3 high-level bullet points don't really follow from the title.  For example, "AI is a great solution for pre-installation validations"?  I recommend moving the first and third bullets (validations and streamlined installation workflow) up to where you discuss "Simplified deployment".  I recommend moving the second bullet up to the beginning of the file where you write "The {ai-full} supports the various deployment platforms with a focus on bare metal."

Author: Done
